### DISCUSSION TITLE
[component] is making IDE slow

### DISCUSSION THREAD

**Discussion by xzbdmw** [leptos-rs/leptos#1484] [Q&A]
When I use the component to mark a function, the CLion I use cannot render colors for methods inside the function, and code completion is also very slow. Does anyone know why? I am using the latest CLion and Rust plugin.

**Comment by sebadob** [comment]
I have no answer how to fully solve this, but it is the same for me with IDEA + Rust Plugin inside the `view!` macros, which is basically the same.
I think this has to do something with the macro unwrapping under the hood.

What actually helped my a little bit and gave better code completion in some places was to activte some experimental Rust macro features inside the IDE:

- press `ctrl + shift + a`
- enter `experimental features`
- actiavate the experimental Rust plugin features for Rust macros

**Reply by xzbdmw** [reply]
yes. view macros works well if there are no #[component] or #[server] above

**Comment by luckynumberke7in** [comment]
Could your issue be related to what Greg mentioned in this post: https://github.com/leptos-rs/leptos/issues/1310 ?

**Reply by xzbdmw** [reply]
This seems to be a common problem. I need to go and see if Intellij Rust has this configuration method.
